frontend/lib/search/filter_bar_button.dart

```diff
@@ -59,10 +60,24 @@ class _FilterBarButtonState extends State<FilterBarButton>
     const paddingPerElement = 8;
     const minNumberElements = 5;
     var totalWidth = MediaQuery.of(context).size.width;
-    var width = min(
-        80.0,
-        (totalWidth - minNumberElements * paddingPerElement) /
-            minNumberElements);
+
+    // Width allowing at least minNumberElements in one row
+    var smallWidth = min(
+      80.0,
+      (totalWidth - minNumberElements * paddingPerElement) / minNumberElements);
+
+    // At least 65 px are needed to fit all category names
+    var largeWidth = max(
+      65.0,
+      smallWidth);
+
+    var numLargeElementsPerRow = totalWidth ~/ largeWidth;
+    var isSecondRow = widget.index >= max(minNumberElements, numLargeElementsPerRow);
+
+    // In the second row we can use larger width as there are only 4 categories
+    var width = isSecondRow
+      ? largeWidth
+      : smallWidth;
```

Comment:
I'm not sure whether this added complexity is worth it. How about refactorign the bar such that we just have two horizontal category lists?

Then we just manually put the elements there and set a custom width for each list. Even with this implementation we will have the magic values 80 and 65.

But I have to say that I do not really get how the items are behaving according to the definition above.

Reply:
So my idea would be to simplify this a little bit by making it even more explicit. We know we have 1-2 rows and we know exactly how wide each icon is.

Reply:
So we have always to rows, one with 5 and one with 4 categories? I think that does not make sense on tablets

Reply:
I'm not sure about which approach is better. On tablets we would ofc add all icons to a single list.

I think the difference is **where** we handle this logic. In this PR it is handled in the button widget. The other approach would handle it at the level of the filter bar widget.

Reply:
Moved magic numbers to constants. Is there a reason why it needs to be handled in the filter bar widget? Otherwise I'd keep it as it works just fine

Reply:
No we can keep it where it is. I just wanted to raise the question whether it can be done in a simpler way, because these min/max formulas are quite complex.

Reply:
Okay, maybe lets wait for a review from @Schedulaar

Reply:
Yeah, the calculations seem quite error prone to me as well, although I don't really have a better suggestion at the top of my head :/

Reply:
Made things more explicit now. I'd merge this as I tested this on multiple devices with reasonable screen size now and did not experience any issues.